Bleed page improvements + pack logic part 2

nikotina91 opened the pull request:
<!-- Add the issues this PR fixes here. If no issues are related to this PR, then this line can be removed. -->
Fixes #1088
Fixes #1263

## Summary of Changes
Did more work on the bleed page, still a work in progress, changes:

- Engine bleed connections (IP/HP) show in amber if there's no pressure available
- HP valves close when engine N2 rpm is slightly above idle, only IP valves remain open 
- Lines on the page should now disappear correctly based on valve position
- "GND" and respective symbol now disappear when in flight
- "APU", APU valve and connection now disappear when the APU is off (? not sure about this one)
- Shown values turn amber when out of range
   - Pack outflow turns amber after 90°
   - Pack inflow turn amber after 230°
   - pressures turn amber when below 4 psi or above 57 psi
   - temperatures turn amber when below 150° or above 270°
   - engine numbers turn amber when N2 below idle and FADEC is energized
- All the temperatures should now be more inline with reality (needs more testing)
- APU bleed temperatures based on APU N2 rpm
- Fault lights now turn on when the packs don't have pressure but the valve is open (issue #1088)
- Packs close during engine start
- Pack are forced to high flow when:
   - TO/GA or MCT/FLX power are applied
   - APU provides bleed
   - Single pack operations
- RAM AIR valve closes when takeoff power is applied
- Pack outlines are now thinner
- "PSI" and "C°" are now in white instead of blue

## Screenshots (if necessary)
All valves open / throttle idle / APU on / on the ground:
![screen1](https://user-images.githubusercontent.com/43645128/95511348-a19cb900-09b7-11eb-918b-efa935831966.png)

Same as before but xbleed is closed:
![screen2](https://user-images.githubusercontent.com/43645128/95511580-f17b8000-09b7-11eb-94a4-62689437d7b8.png)

APU off and xbleed closed:
![screen3](https://user-images.githubusercontent.com/43645128/95511716-21c31e80-09b8-11eb-8146-fa768dba1fde.png)

APU off and xbleed open:
![screen4](https://user-images.githubusercontent.com/43645128/95511813-3f908380-09b8-11eb-96f9-c0d67414211e.png)

xbleed closed, APU powering up:
![screen5](https://user-images.githubusercontent.com/43645128/95512025-88483c80-09b8-11eb-91e9-74da6b499220.png)

eng1 on idle and eng1 bleed open / APU on APU bleed closed:
![screen6](https://user-images.githubusercontent.com/43645128/95512130-ac0b8280-09b8-11eb-8a68-71ba6bc1892c.png)

RAM AIR CLOSED, packs have no pressure + fault lights:
![screen7](https://user-images.githubusercontent.com/43645128/95512295-eaa13d00-09b8-11eb-9e7c-e95c09a9f278.png)
![screen8](https://user-images.githubusercontent.com/43645128/95512366-03a9ee00-09b9-11eb-9b70-d176306dae48.png)

eng 1 bleed on / eng 2 bleed off / apu bleed off / xbleed on / right pack fault:
![screen9](https://user-images.githubusercontent.com/43645128/95512492-30f69c00-09b9-11eb-95cf-1f2e67a3fc7a.png)

engines powering down:
![screen20](https://user-images.githubusercontent.com/43645128/95513759-21785280-09bb-11eb-93c8-542875d27964.png)

fadec powered and N2 below idle:
![fadec_powered_engine_below_idle_amber](https://user-images.githubusercontent.com/43645128/95513785-2d641480-09bb-11eb-8b21-57e2866bfc52.png)

fadec unpowered and N2 below idle:
![fadec_unpowered_eng_below_idle_white](https://user-images.githubusercontent.com/43645128/95513821-3ead2100-09bb-11eb-91f8-00cb9fb995a7.png)

Not sure about this fadec powered/unpowered conditon, will need feedback

## References
cond and packs manuel about the a320: 
http://www.smartcockpit.com/docs/A320-Air_Conditioning_and_Pressurization.pdf

![screen10](https://user-images.githubusercontent.com/43645128/95512758-9cd90480-09b9-11eb-896b-5a285821501e.png)
![screen11](https://user-images.githubusercontent.com/43645128/95512833-b417f200-09b9-11eb-95c2-b0f050f7d03c.png)
![screen12](https://user-images.githubusercontent.com/43645128/95512987-e6c1ea80-09b9-11eb-9a6a-b6dafcb0d6bb.png)
![screen13](https://user-images.githubusercontent.com/43645128/95513053-fccfab00-09b9-11eb-9f89-5dc2826c43fb.png)

http://www.smartcockpit.com/docs/A320-Pneumatic.pdf
![screen14](https://user-images.githubusercontent.com/43645128/95513187-32749400-09ba-11eb-939f-dde2483898fa.png)
![screen15](https://user-images.githubusercontent.com/43645128/95513327-75cf0280-09ba-11eb-94c4-cc412731bb40.png)
![screen16](https://user-images.githubusercontent.com/43645128/95513380-8da68680-09ba-11eb-91d3-96a2827a5790.png)

discord:
![screen17](https://user-images.githubusercontent.com/43645128/95513428-a57e0a80-09ba-11eb-8770-25a1f5c66efd.png)
![screen18](https://user-images.githubusercontent.com/43645128/95513496-bc246180-09ba-11eb-8557-c94730796973.png)
![screen19](https://user-images.githubusercontent.com/43645128/95513549-cf373180-09ba-11eb-9182-cba7a711a75b.png)




## Additional context
It still needs some tuning and probably bug fixing

<!-- You may optionally provide your discord username, so that we may contact you directly about the issue. -->
Discord username (if different from GitHub):

<!-- DO NOT DELETE THIS -->
## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created and uploaded.
The build.py will have already been run with the latest changes, so no need to rerun it once you download the zip.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the right side, slick on the **Artifacts** drop down and click the **A32NX** link

FlyingCarpet7 commented:
Awesome work! Well done!
I will try to give you irl reference  next week on, 1,2- colours, pack ram air operation and som other things. Will take some pictures.

nikotina91 commented:
thanks! that will be very useful!

nikotina91 commented:
Will probably need to change a few things when PR #1238 is merged.

nikotina91 commented:
The pack indicators will only move a few degrees without having the continuous aircond knobs PR as the value of the knob previously were in a 0-2 range and with it they are in a 0-100 range. So please if you are testing this download that PR as well or keep in mind why the indicators will not move much, everything should be fine.

Carter633 commented:
Name: British Avgeek YT - Carter633#1436
Date of testing: Current date 12/10/2020
Version of the sim: 1.9.3.0
PR Tested: #1209

Testing technique: Full flight from Cold and Dark
Steps you've made to test the Pull Request:
1. Spawned at Zurich, Ground power, checked bleed page
2. Started APU - reviewed bleed page throughout
3. Started engines - reviewed bleed page throughout
4. Took off and changed temp settings

Overall Rating: Good
Conclusions of the testing: No issues found

lukecologne commented:
Small problem: The Pack temperature is clipping into the "°C", I don't think that is specific to your PR, but it would be cool if you could fix that. I think it was introduced after the font update, and it looks like the font size may be too big
![grafik](https://user-images.githubusercontent.com/39596827/95776296-5a1e7180-0cc4-11eb-977e-00fc20578ac0.png)

nikotina91 commented:
yeah its the new font, give me a bit and I'll fix it, will also take the chance to add the anti ice indication as suggested by FlyingCarpet7 (with reference pics)

nikotina91 commented:
- Added anti-ice indicators
![ref2](https://user-images.githubusercontent.com/43645128/95784867-f781a180-0cd4-11eb-8e0e-b2fb813fc120.png)

indicators turn amber when there's no bleed supply: https://www.ipadecam.co.uk/#
![ref4](https://user-images.githubusercontent.com/43645128/95785008-3ca5d380-0cd5-11eb-963b-86cd70082208.png)

- fixed pack temperatures clipping into °C

- changed some of the text back to blue (this pic is also from FlyingCarpet7)
![ref1](https://user-images.githubusercontent.com/43645128/95785032-4cbdb300-0cd5-11eb-9723-e7d152ce1045.jpg)

- ram air valve stays close when the pressure difference is more than one psi
![image](https://user-images.githubusercontent.com/43645128/95784859-ef296680-0cd4-11eb-8d14-07ba5b89a348.png)

This is what it looks it like now:
![image](https://user-images.githubusercontent.com/43645128/95785483-22202a00-0cd6-11eb-9385-1ed30e75eec4.png)

lukecologne commented:
I promise, this is my last one: can you reduce the .st9 stroke width a bit (to around 2)? IMO the arc lines and GND triangles are a little to thick

nikotina91 commented:
![image](https://user-images.githubusercontent.com/43645128/95795970-09bb0a00-0cec-11eb-93e3-22b9cf6b70f7.png)

lukecologne reviewed (approved):
Tested and LGTM. I didn't test the systems logic in-depth, since those will hopefully be soon™ implemented in C++.

lukecologne commented on A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_LowerECAM_BLEED.html:
```diff
@@ -6,255 +6,260 @@
-			<text transform="matrix(1 0 0 1 411.1766 214.3244)" class="st1 st2 st6">L</text>
-			<text transform="matrix(1 0 0 1 420.3266 214.3244)" class="st1 st2 st6">O</text>
-			<text transform="matrix(1 0 0 1 543.1766 214.3244)" class="st1 st2 st6">HI</text>
+			<text x="63" y="33" t class="st1 s2 st3">BLEED</text>
```
```suggestion
			<text x="63" y="33" class="st1 s2 st3">BLEED</text>
```
Pretty sure this shouldn't be there, sorry I didn't find it earlier

lukecologne commented on A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_LowerECAM_BLEED.html:
```diff
@@ -6,255 +6,260 @@
 
 <!-- <rect y="0" width="600" height="600" style="fill:url(#ECAMBacklight);"/> -->
 		<g>
```
Either a missing end </g> tag, or the <g> should be removed

lukecologne reviewed (changes requested):
Sorry I didn't find these earlier

lukecologne commented on A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/EICAS/ECAM/A320_Neo_LowerECAM_BLEED.html:
```diff
@@ -6,255 +6,260 @@
 			<!--temp boxes-->
-			<rect x="98.22" y="317.54" class="st10" width="43" height="49" />
-			<rect x="458.72" y="318.04" class="st10" width="43" height="49" />
+			<rect class="st12" x="100" y="317" class="st10" width="42" height="49" />
```
Duplicate class attribute

lukecologne reviewed (approved):
Alright, I fixed the issues I pointed out, and also improved the indentation. I hope this was OK for you.
